apps/lib/apps.c

```diff
@@ -607,9 +608,11 @@ EVP_PKEY *load_keyparams(const char *uri, int maybe_stdin, const char *desc)
 
     (void)load_key_certs_crls(uri, maybe_stdin, NULL, desc,
                               NULL, NULL, &params, NULL, NULL, NULL, NULL);
-    if (params == NULL) {
-        BIO_printf(bio_err, "Unable to load %s\n", desc);
+    if (params != NULL && keytype != NULL && !EVP_PKEY_is_a(params, keytype)) {
+        BIO_printf(bio_err, "Unable to load %s from %s\n", desc, uri);
```

Comment:
Shouldn't this message say that it's not the expected key type?

Reply:
I added some text about this.